src/pages/RFOX/components/Unstake/hooks/useRfoxUnstake.tsx

```diff
@@ -0,0 +1,219 @@
+        !callData ||
+        !adapter
+      )
+        return
+
+      const buildCustomTxInput = await createBuildCustomTxInput({
+        accountNumber: stakingAssetAccountNumber,
+        adapter,
+        data: callData,
+        value: '0',
+        to: RFOX_PROXY_CONTRACT_ADDRESS,
+        wallet,
+      })
+
+      const txId = await buildAndBroadcast({
+        adapter,
+        buildCustomTxInput,
+        receiverAddress: CONTRACT_INTERACTION, // no receiver for this contract call
+      })
+
+      return txId
+    },
+    onSuccess: (txId: string | undefined) => {
+      if (!txId || !setUnstakeTxid) return
+
+      setUnstakeTxid(txId)
+    },
+  })
+
+  const serializedUnstakeTxIndex = useMemo(() => {
+    if (!(unstakeTxid && stakingAssetAccountAddress && stakingAssetAccountId)) return
+
+    return serializeTxIndex(stakingAssetAccountId, unstakeTxid, stakingAssetAccountAddress)
+  }, [stakingAssetAccountAddress, stakingAssetAccountId, unstakeTxid])
+
+  const unstakeTx = useAppSelector(state =>
+    serializedUnstakeTxIndex ? selectTxById(state, serializedUnstakeTxIndex) : undefined,
+  )
+
+  const isUnstakeTxPending = useMemo(
+    () => unstakeMutation.isPending || (unstakeMutation.isSuccess && !unstakeTx),
+    [unstakeMutation.isPending, unstakeMutation.isSuccess, unstakeTx],
+  )
+
+  const isGetUnstakeFeesEnabled = useMemo(
+    () => Boolean(hasEnteredValue && unstakeMutation.isIdle && !Boolean(errors?.amountFieldInput)),
+    [errors?.amountFieldInput, hasEnteredValue, unstakeMutation.isIdle],
+  )
+
+  const unstakeFeesQuery = useEvmFees({
+    ...unstakeFeesQueryInput,
+    enabled: isGetUnstakeFeesEnabled,
+    staleTime: 30_000,
+    // Ensures fees are refetched at an interval, including when the app is in the background
+    refetchIntervalInBackground: true,
+    // Yeah this is arbitrary but come on, Arb is cheap
+    refetchInterval: 15_000,
+  })
+
+  const userStakingBalanceOfQuery = useStakingInfoQuery({
+    stakingAssetAccountAddress,
+    select: ([stakingBalance]) => stakingBalance.toString(),
+  })
+  const { data: userStakingBalanceOfCryptoBaseUnit } = userStakingBalanceOfQuery
+
+  const newContractBalanceOfQuery = useStakingBalanceOfQuery({
+    stakingAssetAccountAddress: RFOX_PROXY_CONTRACT_ADDRESS,
+    stakingAssetId,
+    select: data => data.toString(),
+  })
+  const { data: newContractBalanceOfCryptoBaseUnit } = newContractBalanceOfQuery
+
+  const newShareOfPoolPercentage = useMemo(
+    () =>
+      bnOrZero(userStakingBalanceOfCryptoBaseUnit)
+        .minus(amountCryptoBaseUnit)
+        .div(newContractBalanceOfCryptoBaseUnit ?? 0)
+        .toFixed(4),
+    [amountCryptoBaseUnit, newContractBalanceOfCryptoBaseUnit, userStakingBalanceOfCryptoBaseUnit],
+  )
+
+  return {
+    newShareOfPoolPercentage,
+    isGetUnstakeFeesEnabled,
+    unstakeMutation,
+    unstakeFeesQuery,
+    newContractBalanceOfQuery,
+    userStakingBalanceOfQuery,
+    isUnstakeTxPending,
+  }
```

Comment:
could be worth memoizing this guy

Reply:
Return values is where I would draw the line with "memo everything" 🙏🏽 if we need to, we have bigger problems.
This exposes already memoized values, and react uses shallow comparison - so the only thing using `useMemo` would do here is hindering readibility for paranoia purposes IMO.